theater/start_generator.py

```diff
@@ -43,215 +53,387 @@
+        self.game = game
+        self.control_point = control_point
+        self.templates = templates
 
-        # CP
-        if cp.captured:
-            faction_name = game.player_name
+    @property
+    def faction_name(self) -> str:
+        if self.control_point.captured:
+            return self.game.player_name
         else:
-            faction_name = game.enemy_name
-        faction = db.FACTIONS[faction_name]
-
-        if cp.cptype == ControlPointType.AIRCRAFT_CARRIER_GROUP:
-            # Create ground object group
-            group_id = game.next_group_id()
-            g = TheaterGroundObject("CARRIER")
-            g.group_id = group_id
-            g.object_id = 0
-            g.cp_id = cp.id
-            g.airbase_group = True
-            g.dcs_identifier = "CARRIER"
-            g.sea_object = True
-            g.obj_name = namegen.random_objective_name()
-            g.heading = 0
-            g.position = Point(cp.position.x, cp.position.y)
-            group = generate_carrier_group(faction_name, game, g)
-            g.groups = []
-            if group is not None:
-                g.groups.append(group)
-            cp.ground_objects.append(g)
-            # Set new name :
-            if len(faction.carrier_names) > 0:
-                cp.name = random.choice(faction.carrier_names)
-            else:
-                cp_to_remove.append(cp)
-        elif cp.cptype == ControlPointType.LHA_GROUP:
-            # Create ground object group
-            group_id = game.next_group_id()
-            g = TheaterGroundObject("LHA")
-            g.group_id = group_id
-            g.object_id = 0
-            g.cp_id = cp.id
-            g.airbase_group = True
-            g.dcs_identifier = "LHA"
-            g.sea_object = True
-            g.obj_name = namegen.random_objective_name()
-            g.heading = 0
-            g.position = Point(cp.position.x, cp.position.y)
-            group = generate_lha_group(faction_name, game, g)
-            g.groups = []
-            if group is not None:
-                g.groups.append(group)
-            cp.ground_objects.append(g)
-            # Set new name :
-            if len(faction.helicopter_carrier_names) > 0:
-                cp.name = random.choice(faction.helicopter_carrier_names)
+            return self.game.enemy_name
+
+    @property
+    def faction(self) -> Faction:
+        return db.FACTIONS[self.faction_name]
+
+    def generate(self) -> bool:
+        self.control_point.ground_objects = []
+        self.generate_ground_points()
+        if self.faction.navy_generators:
+            # Even airbases can generate navies if they are close enough to the
+            # water. This is not controlled by the control point definition, but
+            # rather by whether or not the generator can find a valid position
+            # for the ship.
+            self.generate_navy()
+
+        if self.faction.missiles:
+            # TODO: Presumably only for airbases?
```

Comment:
Yes, only for airbase. This is only used for the WW2 german faction (V1 missile site). We could add Scud sites too, but hasn't been done yet.